src/modules/launcher/Wox.Infrastructure/Storage/JsonStorage`1.cs

```diff
@@ -122,17 +125,58 @@ private void BackupOriginFile()
 
         public void Save()
         {
-            try
+            lock (_saveLock)
             {
-                string serialized = JsonSerializer.Serialize(_data, _serializerOptions);
-                File.WriteAllText(FilePath, serialized);
-                _storageHelper.Close();
+                try
+                {
+                    string serialized = JsonSerializer.Serialize(_data, _serializerOptions);
+                    File.WriteAllText(FilePath, serialized);
+                    _storageHelper.Close();
 
-                Log.Info($"Saving cached data at <{FilePath}>", GetType());
+                    Log.Info($"Saving cached data at <{FilePath}>", GetType());
+                }
+                catch (IOException e)
+                {
+                    Log.Exception($"Error in saving data at <{FilePath}>", e, GetType());
+                }
             }
-            catch (IOException e)
+        }
+
+        public void Clear()
+        {
+            if (File.Exists(FilePath))
+            {
+                File.Delete(FilePath);
+                LoadDefault();
+                Log.Info($"Deleting cached data at <{FilePath}>", GetType());
+            }
+        }
+
+        public bool CheckVersionMismatch()
+        {
+            // Skip the fields check if the version hasn't changed.
+            // This optimization prevents unnecessary fields processing when the cache
+            // is already up to date, enhancing performance and reducing IO operations
+            if (!_storageHelper.ClearCache)
+            {
+                return false;
+            }
+
+            _storageHelper.ClearCache = false;
+            return true;
+        }
+
+        public bool CheckFileToUpdateOrClear()
```

Comment:
I'd rename this function to something like `IsFileDataCompatible` or `TryLoadData`